src/lib/pickles/opt_sponge.mli

```diff
@@ -19,10 +19,46 @@ module Make
 
   val create : ?init:Impl.Field.t array -> Impl.Field.t Sponge.Params.t -> t
 
+  (** Create a new sponge with state copied from the given sponge.
+      In particular, this copies the underlying state array, so that any
+      mutations to the copy will not affect the original.
+  *)
+  val copy : t -> t
+
   val of_sponge : Impl.Field.t Sponge.t -> t
 
   val absorb :
     t -> Impl.Field.t Snarky_backendless.Boolean.t * Impl.Field.t -> unit
 
   val squeeze : t -> Impl.Field.t
+
+  (** Updates the sponge state by forcing absorption of all 'pending' field
+      elements passed to [absorb].
+      This method runs logic equivalent to that in the [squeeze] method, but
+      without transitioning the state.
+      This method can be used with [copy] to create a fork of a sponge where
+      one of the branches calls the [absorb] method and the other does not.
+  *)
+  val consume_all_pending : t -> unit
+
+  (** Recombines a forked copy of a sponge with the original.
+      When the boolean value is true, the sponge state will be preserved;
+      otherwise it will be overwritten by the state of the original sponge.
+
+      When an optional [squeeze] has ocurred, both the original and forked
+      sponges have called [consume_all_pending] before the squeeze, and must
```

Comment:
If you want to refer to the function `consume_all_pending` and have a link while generating the doc with odoc, you can use `{!consume_all_pending}`.